# DLCP

DLCP is a deep learning - based procedure to stratify cancer patients, identify the possible clinical prognosis biomarkers and their small molecular ligands.

DLCP includes three steps:

1. performs icluster by integrating integrated genomic, transcriptomic and epigenetic data and cancer survival (DNN-c.py)

2. stratifies cancer patients cancer patients and identifies clinical prognosis biomarkers that closely related with cancer survival and tumor progression (identpatS.R)

4. screens small molecular ligands for cancer treatment (CIPHEN https://github.com/wangyc82/CIPHEN)


The requirements of DNN includes numpy, torch, torch.nn and torch.optim. The input data for training DNN includes cancer omics represented by matrices with gene as row and patient as column. The output was the survival vectors that were generated by survival status and last follow-up time. Using DNN-c.py with cancer omics matrices as input and survival vectors as output to learn the parameters $w_l$ (l=3) of DNN with one input, two hidden, and one output layer, and the significance of ith input feature in predicting cancer survival ($s_i$) was evaluated by aggregating the weights assigned to that feature across all layers of the deep learning model, that is $s_i=\sum_{l} w_l^i$. The top 50 features were selected for further analysis using R code.

The R codes for signature analysis required package of pheatmap to perform hierarchical clustering and capture the diverse profiles of signatures in high and low risk patients, ggpubr to display the differences in signatures between high and low risk patients. The differentially expressed and methylated analysis were performed by limma package
